src/main/scala/tech/cryptonomic/conseil/routes/Data.scala

```diff
@@ -35,27 +35,26 @@ class Data(config: PlatformsConfiguration, queryProtocolPlatform: DataPlatform,
 
   import cats.instances.either._
   import cats.instances.future._
-  import cats.instances.option._
   import cats.syntax.bitraverse._
-  import cats.syntax.traverse._
 
   /** V2 Route implementation for query endpoint */
   val postRoute: Route = queryEndpoint.implementedByAsync {
     case ((platform, network, entity), apiQuery, _) =>
-      apiQuery.validate(EntityPath(entity, NetworkPath(network, PlatformPath(platform))), metadataService).flatMap {
-        validationResult =>
+      val path = EntityPath(entity, NetworkPath(network, PlatformPath(platform)))
+
+      pathValidation(path) {
+        apiQuery.validate(path, metadataService).flatMap { validationResult =>
           validationResult.map { validQuery =>
-            platformNetworkValidation(platform, network) {
-              queryProtocolPlatform.queryWithPredicates(platform, entity, validQuery).map { queryResponseOpt =>
-                queryResponseOpt.map { queryResponses =>
-                  QueryResponseWithOutput(
-                    queryResponses,
-                    validQuery.output
-                  )
-                }
+            queryProtocolPlatform.queryWithPredicates(platform, entity, validQuery).map { queryResponseOpt =>
```

Comment:
I can't see platformNetworkValidation being done anymore? Is that intended?

Reply:
It's done a few lines above as `pathValidation(path)`.